app/assets/javascripts/directives/bDropzone.coffee

```diff
@@ -23,7 +25,75 @@
         event.preventDefault()
       (event.originalEvent or event).dataTransfer.effectAllowed = 'move'
       false
-    validMimeTypes = attrs.bDrop
+
+    # handle files from datatransfer files
+    addFile = (file) ->
+      obj = {}
+      obj.name = file.name
+      obj.type = file.type
+      reader = new FileReader()
+      
+      reader.onload = (evt) ->
+        # update bindings
+        scope.$apply ->
+          obj.data = reader.result.split(",")[1]
+          scope.drop = true
+          scope.info = ''
+          scope.theFiles.push obj
+
+          if file.type in ['image/jpeg', 'image/png', 'image/tiff', 'image/gif']
+            scope.images.push evt.target.result
+          return
+
+      if file.type != ''
+        reader.readAsDataURL file
+      else
+        scope.$apply ->
+          scope.noType = scope.noType + file.name + " has no type and can not be saved"
+      return false
+
+    # handle files from directory?
+    # does not support wierder types such as .cpp and .xcf
+    addFileFromDir = (entry) ->
+      entry.file((file)->
```

Comment:
I don't really know how it workds but for every entry there is a built in funtion file to retrieve the file from the entry. It is a native function and not the same as addFile.